backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java

```diff
@@ -506,6 +509,29 @@ public void updateDescriptionFromDocType(final I_C_Order order)
 		}
 	}
 
+	@ModelChange(timings = {
+			ModelValidator.TYPE_BEFORE_NEW,
+			ModelValidator.TYPE_BEFORE_CHANGE
+	}, ifColumnsChanged = {
+			I_C_Order.COLUMNNAME_C_DocTypeTarget_ID
+	})
+	public void removeFlatRateConditionsForCallOrder(final I_C_Order order)
+	{
+		if (order.getC_DocTypeTarget_ID() <= 0)
+		{
+			return;
+		}
+		final I_C_DocType dt = Services.get(IDocTypeDAO.class).getById(order.getC_DocTypeTarget_ID());
```

Comment:
please, move this at the beginning of the class with the rest of the services.

Reply:
done